frontend/src/components/data-table/renderers.tsx

```diff
@@ -178,3 +178,23 @@ function columnSizingHandler<TData>(
     [column.id]: thead.getBoundingClientRect().width,
   }));
 }
+
+/**
+ * Render an unknown value as a string. Converts objects to JSON strings.
+ * @param opts.value - The value to render.
+ * @param opts.nullAsEmptyString - If true, null values will be "". Else, stringify.
+ */
+export function renderUnknownValue(opts: {
+  value: unknown;
+  nullAsEmptyString?: boolean;
+}) {
```

Comment:
```suggestion
}): string {
```

super nit: but just because it's used downstream expecting a string